app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java

```diff
@@ -218,7 +220,27 @@ public void deleteGroceryList(View view) {
     }
 
     public void deleteTaskList(View view) {
-        // TODO : The task list is not linked to households yet
+        OnFailureListener tlDeletionFailed =
+                exception -> Toast.makeText(getApplicationContext(),
+                        "Cannot remove task list", Toast.LENGTH_SHORT).show();
+
+        firestore.collection("task_lists")
+                .whereEqualTo("hh-id", currentHousehold.getId())
+                .get()
+                .addOnSuccessListener(docRefList -> {
+                    assert docRefList.getDocuments().size() == 1;
+
+                    DocumentSnapshot metadataSnap = docRefList.getDocuments().get(0);
+
+                    List<DocumentReference> taskPtrs = (ArrayList<DocumentReference>)
+                            metadataSnap.getData().getOrDefault("task-ptrs", new ArrayList<>());
+
+                    assert taskPtrs != null;
+                    taskPtrs.forEach(taskDocRef -> taskDocRef.delete().addOnFailureListener(tlDeletionFailed));
+
+                    metadataSnap.getReference().delete().addOnFailureListener(tlDeletionFailed);
```

Comment:
Is it a problem if the metadata is deleted before the taskPtr ? If not there is no problem but otherwise it might be better to delete it in the TaskPtr onFailureListener

Reply:
There is one case where we'd have a problem: if there's a taskPtr that can't be deleted, but we deleted the metadata, we have no idea which one it could have been. We have like 3 solutions:
-  schedule a periodical cleanup of non-linked tasks from task_dump
- log the taskPtrs that couldn't be deleted in a special file on firebase & have someone delete them manually
- delete the metadata if and only if all the taskPtrs were successfully deleted

Reply:
Thanks for the reflexion and the fix !